packages/bun-plugin-svelte/src/index.ts

```diff
@@ -0,0 +1,91 @@
+import type { BunPlugin, BuildConfig, OnLoadResult } from "bun";
+import { strict as assert } from "node:assert";
+import { basename } from "node:path";
+import { compile, compileModule } from "svelte/compiler";
+import { getBaseCompileOptions, validateOptions, type SvelteOptions, hash } from "./options";
+
+const kEmptyObject = Object.create(null);
+const virtualNamespace = "bun-svelte";
+
+function SveltePlugin(options: SvelteOptions = kEmptyObject as SvelteOptions): BunPlugin {
+  if (options != null) validateOptions(options);
+
+  /**
+   * import specifier -> CSS source code
+   */
+  const virtualCssModules = new Map<string, string>();
+
+  return {
+    name: "bun-plugin-svelte",
+    setup(builder) {
+      const { config = kEmptyObject as Partial<BuildConfig> } = builder;
+      const baseCompileOptions = getBaseCompileOptions(options ?? (kEmptyObject as Partial<SvelteOptions>), config);
+
+      builder
+        .onLoad({ filter: /\.svelte(?:\.[tj]s)?$/ }, async args => {
+          assert(args && typeof args === "object");
```

Comment:
these asserts do not need to run for all our users. instead of here, these assertions belong in `test/bundler/bundler_plugin.test.ts`